packages/ui5-application-writer/src/data/defaults.ts

```diff
@@ -120,22 +120,20 @@ export function mergeUi5(ui5: Partial<UI5>): UI5 {
 }
 
 /**
- * Get the best types version for the given minUI5Version.
+ * Get the best types version for the given minUI5Version within a selective range, starting at 1.90.0 as the minimum
  * For the latest versions the LTS S/4 on-premise version (1.102.x) is used, for anything before we match the versions as far back as available.
  *
- * @param minUI5Version the mininum UI5 version that needs to be supported
+ * @param minUI5Version the minimum UI5 version that needs to be supported
  * @returns semantic version representing the types version.
  */
 export function getTypesVersion(minUI5Version?: string) {
     const version = semVer.coerce(minUI5Version);
-    if (!version) {
-        return `~${UI5_DEFAULT.TYPES_VERSION_BEST}`;
-    } else if (semVer.gte(version, UI5_DEFAULT.TYPES_VERSION_BEST)) {
+    if (!version || semVer.gte(version, UI5_DEFAULT.TYPES_VERSION_BEST)) {
         return `~${UI5_DEFAULT.TYPES_VERSION_BEST}`;
     } else {
         return semVer.gte(version, UI5_DEFAULT.TYPES_VERSION_SINCE)
-            ? `~${semVer.major(version)}.${semVer.minor(version)}.${semVer.patch(version)}`
-            : UI5_DEFAULT.TYPES_VERSION_PREVIOUS;
+            ? `~${semVer.major(version)}.${semVer.minor(version)}.0`
+            : `~${UI5_DEFAULT.TYPES_VERSION_PREVIOUS}`;
```

Comment:
Any reason why we are using 0 and not the patch version provided?

Reply:
Since we have no guarantee the minUI5version will match the corresponding types versions, its a safety net and future patch versions will be used since the tilda is also specified.

Reply:
Ok, good point, using internal knowledge, we would know that the release of a types version is bound to a release of a UI5 version, so they always exist together, however, as you said, due to the ~ it doesn't make a difference.